src/main/java/ua/team3/carsharingservice/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package ua.team3.carsharingservice.service.impl;
+
+import java.util.Collections;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import ua.team3.carsharingservice.dto.UserRegistrationRequestDto;
+import ua.team3.carsharingservice.dto.UserResponseDto;
+import ua.team3.carsharingservice.dto.UserUpdateRequestDto;
+import ua.team3.carsharingservice.exception.RegistrationException;
+import ua.team3.carsharingservice.mapper.UserMapper;
+import ua.team3.carsharingservice.model.Role;
+import ua.team3.carsharingservice.model.User;
+import ua.team3.carsharingservice.repository.RoleRepository;
+import ua.team3.carsharingservice.repository.UserRepository;
+import ua.team3.carsharingservice.service.UserService;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final RoleRepository roleRepository;
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserResponseDto save(UserRegistrationRequestDto requestDto
+    ) throws RegistrationException {
+        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

Comment:
```suggestion
        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

You can reuse here findByEmail(). No need have existByEmail() i think

Reply:
Leave as is